rcgen/src/lib.rs

```diff
@@ -619,17 +611,10 @@ impl CertificateParams {
 	/// for the presence of the `BasicConstraints` extension, or perform any other
 	/// validation.
 	#[cfg(feature = "x509-parser")]
-	pub fn from_ca_cert_der(ca_cert: &[u8], key_pair: KeyPair) -> Result<Self, Error> {
+	pub fn from_ca_cert_der(ca_cert: &[u8]) -> Result<Self, Error> {
```

Comment:
I think the rustdoc comments on this fn need to be tweaked for the change in return type. Presently it says:
> This function is only of use if you have an existing CA certificate you would like to use to sign a certificate generated by rcgen. By providing the returned [Certificate](https://github.com/rustls/rcgen/pull/213/struct.Certificate.html) and the **passed-through [KeyPair](https://github.com/rustls/rcgen/pull/213/struct.KeyPair.html)**

Reply:
> I didn't find the notion of "passed-through" very clear. 

Yeah, that (awkward) phrasing was added by me in #212 but looking at the diff again I think it was a mistake. The `CertificateParms::from_ca_cert_der` fn only returns `CertificateParams` and didn't pass through the keypair. I was probably thinking of the `Certificate::generate*` APIs that return both a `Certificate` and `KeyPair`, and for the case where you specify a `KeyPair` in the `CertificateParams` it is "passed through" to the caller in the return args (vs generating one from scratch). All of that falls away with this changeset so we can lose the awkward "pass through" notion entirely.

I think the right fix for this branch is to replace this text:
>  By providing the returned [`Certificate`] and the passed-through [`KeyPair`] to [`Certificate::generate`] or  [`Certificate::from_request`] functions you can generate a new certificate from [`CertificateParams`], signed by the pre-existing CA.

with something like:
> By providing the constructed [`CertificateParams`] and the [`KeyPair`] associated with your existing `ca_cert` you can use [`Certificate::generate`] or [`Certificate::from_request`] to issue new certificates using the CA cert.

> There doesn't seem to be a lot of context in https://github.com/rustls/rcgen/pull/12.

I'm also a bit skeptical of this API surface. I feel like there's some lurking danger in the way rcgen tries to reconstitute `CertificateParams` from the DER, only puling out a handful of parameters deemed needed for signing :-/

Reply:
I've adopted your language proposal. 👍